engine/lib/phx/src/system/guid.rs

```diff
@@ -2,17 +2,21 @@ use std::sync::atomic::{AtomicU64, Ordering};
 
 static NEXT_ID: AtomicU64 = AtomicU64::new(1);
 
-#[no_mangle]
-pub extern "C" fn GUID_Create() -> u64 {
-    NEXT_ID.fetch_add(1, Ordering::Relaxed)
+#[repr(C)]
+pub struct GUID {
 }
 
-#[no_mangle]
-pub extern "C" fn GUID_Exists(id: u64) -> bool {
-    id < NEXT_ID.load(Ordering::Relaxed) && id != 0
-}
+#[luajit_ffi_gen::luajit_ffi]
+impl GUID {
+    pub fn create() -> u64 {
+        NEXT_ID.fetch_add(1, Ordering::Relaxed)
+    }
 
-#[no_mangle]
-pub extern "C" fn GUID_Reset() {
-    NEXT_ID.store(1, Ordering::Relaxed);
-}
+    pub fn exists(id: u64) -> bool {
+        id < NEXT_ID.load(Ordering::Relaxed) && id != 0
+    }
+
+    pub fn reset() {
+        NEXT_ID.store(1, Ordering::Relaxed);
+    }
+}
```

Comment:
By default we put a new line at the end of the file `cargo fmt` or vscode flag will do this for you.